test/voracle/TestLag.sol

```diff
@@ -0,0 +1,193 @@
+pragma solidity ^0.5.0;
+
+import "../../contracts/voracle/Lag.sol";
+import "../mocks/MockIPRS.sol";
+import "truffle/Assert.sol";
+
+contract TestLag {
+    Lag public lag;
+    Lag public lagNoneGov;
+    MockIPRS public mockIPRS;
+    MockIPRS public mockIPRS2;
+
+    function beforeEach() public {
+        mockIPRS = new MockIPRS(10);
+        mockIPRS2 = new MockIPRS(11);
+        lag = new Lag(address(this), address(0));
+        lagNoneGov = new Lag(address(1), address(2));
+    }
+
+    function testHalt() public {
+        lag.halt();
+
+        Assert.isTrue(lag.halted(), "lag.halted are should be true");
+    }
+
+    function testHaltWhenNotGov() public {
+        (bool r,) = address(lagNoneGov).call(abi.encodePacked(lagNoneGov.halt.selector));
+
+        Assert.equal(r, false, "lag.halted are should be false");
+    }
+
+    function testResume() public {
+        lag.resume();
+
+        Assert.equal(lag.halted(), false, "lag.halted are should be false");
+    }
+
+    function testSetPriceRefStorage() public {
+        address expectedPriceRefStorage = address(mockIPRS);
+        lag.setPriceRefStorage(address(mockIPRS));
+
+        Assert.equal(lag.priceRefStorage(), expectedPriceRefStorage, "lag.setPriceRefStorage are should be 0xb04aD816e86bFA5515c35Ad02081F71D9E848C88");
+    }
+
+    function testSetLagTime() public {
+        bytes4 selector = lag.setLagTime.selector;
+        uint16 newLagTime = 60 * 60 * 16;
+        bytes memory data = abi.encodeWithSelector(selector, newLagTime);
+        (bool r,) = address(lag).call(data);
+
+        Assert.isTrue(r, "lag.lagTime should be no error");
+    }
+
+    function testSetLagTimeWithError() public {
+        bytes4 selector = lag.setLagTime.selector;
+        uint16 newLagTime = 0;
+        bytes memory data = abi.encodeWithSelector(selector, newLagTime);
+        (bool r,) = address(lag).call(data);
+
+        Assert.isFalse(r, "lag.lagTime should be error");
+    }
+
+    function testVoid() public {
+        lag.void();
+
+        Assert.equal(lag.halted(), true, "lag.void variable halted should be true");
```

Comment:
assert massage not makes sense. something like "lag.halted() should be true after lag.void() has been called"